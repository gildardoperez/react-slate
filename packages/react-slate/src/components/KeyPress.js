/* @flow */

import React from 'react';
import readline from 'readline';

type ReadableStream = stream$Readable & {
  setRawMode(flag: boolean): void,
};

type Key = {
  name: string,
  ctrl: boolean,
  shift: boolean,
  alt: boolean,
};

type Props = {
  stream: ReadableStream,
  onPress(char: string, key: Key): void,
  children?: any,
};

export default class KeyPress extends React.Component<Props> {
  isStreamConfigured: boolean = false;

  onKeyPress = (ch: string, key: Key) => {
    this.props.onPress(ch, key);
    if (key.ctrl && key.name === 'c') {
      process.exit(0);
    }
  };

  componentDidMount() {
    const { stream } = this.props;
    if (!this.isStreamConfigured) {
      stream.setRawMode(true);
      readline.emitKeypressEvents(stream);
      this.isStreamConfigured = true;
    }
    stream.addListener('keypress', this.onKeyPress);
  }

  componentWillUnmount() {
    const { stream } = this.props;
    stream.setRawMode(false);
    stream.removeListener('keypress', this.onKeyPress);
  }

  render() {
    return this.props.children || null;
  }
}
