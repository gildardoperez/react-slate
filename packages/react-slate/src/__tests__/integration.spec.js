/* @flow */

import React from 'react';
import { View } from '../components';
import renderToString from '../public/renderToString';

describe('in integration tests renderToString', () => {
  it('should apply backgroundColor to paddings', () => {
    expect(
      renderToString(
        <View style={{ padding: '2', backgroundColor: 'ansi-red' }}>
          Hello World
        </View>,
        {
          height: 6,
        }
      )
    ).toMatchSnapshot();
  });

  it('should apply backgroundColor whole content with fixed height', () => {
    expect(
      renderToString(
        <View style={{ height: 4, backgroundColor: 'ansi-red' }}>
          Hello World
        </View>,
        {
          height: 6,
        }
      )
    ).toMatchSnapshot();

    expect(
      renderToString(
        <View style={{ height: 4, width: 12, backgroundColor: 'ansi-red' }}>
          Hello World
        </View>,
        {
          height: 6,
        }
      )
    ).toMatchSnapshot();
  });
});
