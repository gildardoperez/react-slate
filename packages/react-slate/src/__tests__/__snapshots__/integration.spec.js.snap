// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in integration tests renderToString should apply backgroundColor to paddings 1`] = `
"@[41m               @[49m                         
@[41m               @[49m                         
@[41m  Hello World  @[49m                         
@[41m               @[49m                         
@[41m               @[49m                         
                                        "
`;

exports[`in integration tests renderToString should apply backgroundColor whole content with fixed height 1`] = `
"@[41mHello World@[49m                             
@[41m           @[49m                             
@[41m           @[49m                             
@[41m           @[49m                             
                                        
                                        "
`;

exports[`in integration tests renderToString should apply backgroundColor whole content with fixed height 2`] = `
"@[41mHello World @[49m                            
@[41m            @[49m                            
@[41m            @[49m                            
@[41m            @[49m                            
                                        
                                        "
`;
